src/compiler/compile/render_dom/index.ts

```diff
@@ -49,11 +49,8 @@ export default function dom(
 
 	if (should_add_css) {
 		body.push(b`
-			function ${add_css}() {
-				var style = @element("style");
-				style.id = "${component.stylesheet.id}-style";
-				style.textContent = "${styles}";
-				@append(@_document.head, style);
+			function ${add_css}(target) {
+				@append_styles(target, "${component.stylesheet.id.replace('svelte-', '')}", "${styles}");
```

Comment:
This also was implemented to save additional 14 bytes per component.
Thanks for pointing it out. I just realized that since a few months there exists the functionality to apply custom style-hashes.
In the current implementation the functionality would not be compromised, but the hash would be wrong.
I will fix the implementation 👍